ffi_derive/src/lib.rs

```diff
@@ -212,3 +279,16 @@ fn impl_ffi_macro(ast: DeriveInput) -> TokenStream {
     }
     .into()
 }
+
+/// Parses a module that contains typealiases and stores that information for other ffi_derive calls
+/// to use later in resolving aliases.
+///
+#[proc_macro_attribute]
+pub fn alias_resolution(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let resolution_key = attr.to_string();
+    let module: syn::ItemMod = syn::parse(item.clone())
```

Comment:
Not a big deal for a procmacro, but I tend to like to make the caller `clone()`, so there's no surprises. Don't feel obligated to change though.

Reply:
Yeah, I think it's clearer when the caller does the `clone` too. I'm not sure I'm following where we'd have the caller do a `clone()` here though -- `item` gets moved into this macro function (by the `#[alias_resolution]` macro invocation); I'm only doing a `clone()` when calling `syn::parse()` because I wanted to be able to also dump the module into the panic message below on failure (which is overkill and should just report an error to the appropriate span, but for now it helps with debugging).

Reply:
Right, those `TokenStream`s just get passed to the macro when the compiler invokes it.